test/functional/wallet_fundrawtransaction.py

```diff
@@ -1312,6 +1313,38 @@ def test_weight_calculation(self):
 
         self.nodes[2].unloadwallet("test_weight_calculation")
 
+    def test_weight_limits(self):
+        self.log.info("Test weight limits")
+
+        self.nodes[2].createwallet("test_weight_limits")
+        wallet = self.nodes[2].get_wallet_rpc("test_weight_limits")
+
+        outputs = []
+        for _ in range(1472):
+            outputs.append({wallet.getnewaddress(address_type="legacy"): 0.1})
+        txid = self.nodes[0].send(outputs=outputs)["txid"]
+        self.generate(self.nodes[0], 1)
+
+        # 272 WU per input (273 when high-s); picking 1471 inputs will exceed the max standard tx weight.
+        rawtx = wallet.createrawtransaction([], [{wallet.getnewaddress(): 0.1 * 1471}])
+
+        # 1) Try to fund transaction only using the preset inputs
+        input_weights = []
+        for i in range(1471):
+            input_weights.append({"txid": txid, "vout": i, "weight": 273})
+        assert_raises_rpc_error(-4, "Transaction too large", wallet.fundrawtransaction, hexstring=rawtx, input_weights=input_weights)
+
+        # 2) Let the wallet fund the transaction
+        assert_raises_rpc_error(-4, "The inputs size exceeds the maximum weight. Please try sending a smaller amount or manually consolidating your wallet's UTXOs",
+                                wallet.fundrawtransaction, hexstring=rawtx)
+
+        # 3) Pre-select some inputs and let the wallet fill-up the remaining amount
+        inputs = input_weights[0:1000]
+        assert_raises_rpc_error(-4, "The combination of the pre-selected inputs and the wallet automatic inputs selection exceeds the transaction maximum weight. Please try sending a smaller amount or manually consolidating your wallet's UTXOs",
```

Comment:
Tested that this would fail if the `return util::Error...` line was commented out in source/wallet/spend.cpp.  It did (as expected).

Reply:
Re-ran this test (comment out `return util::Error...`) for 72b226882fe2348a9a66aee1d8d21b4e2d275e68.  Failed as expected.